massa-execution-worker/src/execution.rs

```diff
@@ -877,7 +801,12 @@ impl ExecutionState {
             .active_cursor
             .get_next_slot(self.config.thread_count)
             .expect("slot overflow when getting speculative ledger");
-        let search_result = self.fetch_active_history_balance(next_slot, address);
+        self.verify_active_slot(next_slot);
+        let index = self.get_active_index(next_slot);
```

Comment:
this will always return None because "next_slot" is by definition one step beyond the end of history (see comment above)

Reply:
Used `active_cursor` instead.

Reply:
I'm still not sure about this, will think more about it tomorrow